client/my-sites/site-settings/site-icon-setting/index.jsx

```diff
@@ -105,7 +106,13 @@ class SiteIconSetting extends Component {
 			}
 
 			MediaStore.off( 'change', checkUploadComplete );
-			this.saveSiteIconSetting( siteId, media );
```

Comment:
>Should we avoid calling `this.props.receiveMedia( siteId, media )` above if we have no media?

Good question. I recall I'd recently moved that line to before the `isItemBeingUploaded` probably to allow optimistic display of the image while it's uploading. Let me check what the impact here is and what alternatives we have...